src/java.prefs/macosx/native/libprefs/MacOSXPreferencesFile.m

```diff
@@ -682,7 +682,7 @@ static void createTreeForPath(CFStringRef path, CFStringRef name,
     CFDictionaryRef node;
     CFStringRef topKey;
     CFMutableDictionaryRef topValue;
-    Boolean beforeAdd = false;
```

Comment:
The return value from `CFDictionaryContainsKey` is a Boolean and is assigned to this variable. So I think these changes are the wrong way round. Keep this as a Boolean but convert the return value to jboolean:
```
return beforeAdd ? JNI_TRUE : JNI_FALSE;
```

Reply:
Argh, looks like Apple and Objective-C still use the ancient Boolean types from Carbon for the CoreFoundation APIs. Sigh...